src/components/TodoItem.js

```diff
@@ -0,0 +1,90 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+
+export class TodoItem extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      isEditing: false,
+    };
+    this.editingInput = React.createRef();
+  }
+
+  componentDidUpdate() {
+    if (this.state.isEditing) {
+      this.editingInput.current.focus();
+    }
+  }
+
+  handleTextInputChange = ({ target }) => {
+    const { value } = target;
+
+    this.props.handleEditTodo(this.props.index, value);
+  }
+
+  toggleEditing = () => {
+    if (this.props.todo.text === '') {
+      return this.editingInput.current.focus();
```

Comment:
it is strange to return focus result
```suggestion
      this.editingInput.current.focus();
      return;
```
Will be great to show some error message